sys/tiny_strerror/tiny_strerror.c

```diff
@@ -0,0 +1,132 @@
+    [EBUSY]             = "-EBUSY",
+    [ECANCELED]         = "-ECANCELED",
+    [ECHILD]            = "-ECHILD",
+    [ECONNABORTED]      = "-ECONNABORTED",
+    [ECONNREFUSED]      = "-ECONNREFUSED",
+    [ECONNRESET]        = "-ECONNRESET",
+    [EDEADLK]           = "-EDEADLK",
+    [EDESTADDRREQ]      = "-EDESTADDRREQ",
+    [EDOM]              = "-EDOM",
+    [EEXIST]            = "-EEXIST",
+    [EFAULT]            = "-EFAULT",
+    [EFBIG]             = "-EFBIG",
+    [EHOSTUNREACH]      = "-EHOSTUNREACH",
+    [EILSEQ]            = "-EILSEQ",
+    [EINPROGRESS]       = "-EINPROGRESS",
+    [EINTR]             = "-EINTR",
+    [EINVAL]            = "-EINVAL",
+    [EIO]               = "-EIO",
+    [EISCONN]           = "-EISCONN",
+    [EISDIR]            = "-EISDIR",
+    [ELOOP]             = "-ELOOP",
+    [EMFILE]            = "-EMFILE",
+    [EMLINK]            = "-EMLINK",
+    [EMSGSIZE]          = "-EMSGSIZE",
+    [ENAMETOOLONG]      = "-ENAMETOOLONG",
+    [ENETDOWN]          = "-ENETDOWN",
+    [ENETRESET]         = "-ENETRESET",
+    [ENETUNREACH]       = "-ENETUNREACH",
+    [ENFILE]            = "-ENFILE",
+    [ENOBUFS]           = "-ENOBUFS",
+    [ENODEV]            = "-ENODEV",
+    [ENOENT]            = "-ENOENT",
+    [ENOEXEC]           = "-ENOEXEC",
+    [ENOLCK]            = "-ENOLCK",
+    [ENOMEM]            = "-ENOMEM",
+    [ENOMSG]            = "-ENOMSG",
+    [ENOPROTOOPT]       = "-ENOPROTOOPT",
+    [ENOSPC]            = "-ENOSPC",
+    [ENOSYS]            = "-ENOSYS",
+    [ENOTCONN]          = "-ENOTCONN",
+    [ENOTDIR]           = "-ENOTDIR",
+    [ENOTEMPTY]         = "-ENOTEMPTY",
+    [ENOTSOCK]          = "-ENOTSOCK",
+    [ENOTSUP]           = "-ENOTSUP",
+    [ENOTTY]            = "-ENOTTY",
+    [ENXIO]             = "-ENXIO",
+    [EPERM]             = "-EPERM",
+    [EPIPE]             = "-EPIPE",
+    [EPROTONOSUPPORT]   = "-EPROTONOSUPPORT",
+    [EPROTOTYPE]        = "-EPROTOTYPE",
+    [ERANGE]            = "-ERANGE",
+    [EROFS]             = "-EROFS",
+    [ESPIPE]            = "-ESPIPE",
+    [ESRCH]             = "-ESRCH",
+    [ETIMEDOUT]         = "-ETIMEDOUT",
+    [EXDEV]             = "-EXDEV",
+    /* EAGAIN and EWOULDBLOCK have the exact same meaning and consequently may
+     * have the same numeric value */
+#if EAGAIN != EWOULDBLOCK
+    [EWOULDBLOCK]       = "-EWOULDBLOCK",
+#endif
+    /* ENOTSUP and EOPNOTSUPP do not have the exact same meaning. Still, they
+     * have the same numeric value on Linux, breaking POSIX standard */
+#if ENOTSUP != EOPNOTSUPP
+    [EOPNOTSUPP]        = "-EOPNOTSUPP",
+#endif
+};
+
+const char *tiny_strerror(int errnum)
+{
+    /* dark magic: All error strings start with a "-". For positive error codes
+     * an offset of 1 is added to the address of the string, jumping one char
+     * behind the "-". This way the strings do not have to be allocated twice
+     * (once with and once without minus char).
```

Comment:
Ah here it is explained! Thanks.

Reply:
The `vfs` shell command's errno2str implementation previously had support for indicating negative and positive `errno` constants. I didn't want this to feel like a regression.

(And I did indeed ran into an `errno` constant not covered by the `vfs` commend that I had to look up, which triggered this PR.)